app/models/manageiq/providers/redhat/infra_manager/ovirt_services.rb

```diff
@@ -1054,10 +1055,14 @@ def local_external_vnic_profiles
     end
 
     def all_networks_as_switches_in_cluster(uid_ems_cluster)
-      cluster = EmsCluster.find_by(:uid_ems => uid_ems_cluster)
+      cluster = cluster_by_uid_ems(uid_ems_cluster)
       cluster&.switches
     end
 
+    def cluster_by_uid_ems(uid_ems)
+      EmsCluster.find_by(:uid_ems => uid_ems)
```

Comment:
Do you have access to the ems_id from here?  We shouldn't be looking up records by just uid_ems since they aren't guaranteed to be unique.

Ideally this would be `ext_management_system.ems_clusters.find_by(:uid_ems => uid_ems_cluster)`

Reply:
I realize this is just a move but it is a good chance to fix an existing bug

Reply:
I agree, I refactored to avoid duplication. I'll change it right now.

Reply:
:+1: looks good now